main.go

```diff
@@ -104,15 +115,71 @@ func initLogging() (*os.File, error) {
 	return file, nil
 }
 
+// extension specific PublicSettings
+type PublicSettings struct {
+	Script   string   `json:"script"`
+	FileURLs []string `json:"fileUris"`
+}
+
+// extension specific ProtectedSettings
+type ProtectedSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
+	StorageAccountKey  string   `json:"storageAccountKey"`
+}
+
 func main() {
-	// TODO Add more robust error handling
 	file, err := initLogging()
 	if err != nil {
 		fmt.Println("Error opening the provided logfile.")
-		os.Exit(1)
+		os.Exit(-1)
 	}
 	defer file.Close()
```

Comment:
I actually did not catch this earlier, but why are you not closing the file in the `initLogging()` function itself? Does the file have to be open for the logger to work properly?

Reply:
Yes, it has to be open by my understanding

Reply:
Sure that makes sense. But why do you have 2 `file.close()` statements? One in line 129 and the other here?